src/components/filter/filter-comp/ratings-component/ratings-filter.js

```diff
@@ -1,24 +1,37 @@
 import "./ratings.css"
-
+import { useSortedData } from "../../../../context/filter-context"
 export function RatingsFilter() {
+    const { Pratings, dispatch } = useSortedData()
     return (
-        <div class="sort-by-rating">
-            <span class="sort-heading">Sort by ratings</span>
+        <div className="sort-by-rating">
+            <span className="sort-heading">Sort by ratings</span>
             <div className="ratings-filter-div">
                 <label htmlFor="rating-input">
-                    <input className="rating-input" type={"radio"}></input>
+                    <input className="rating-input" type={"radio"} name="rate"
+                    onChange={() => dispatch({ type: "RATINGS", payload: "4_above" })}
+                    checked={Pratings}
+                    ></input>
                     4 stars and above
                 </label>
                 <label htmlFor="rating-input">
-                    <input className="rating-input" type={"radio"}></input>
+                    <input className="rating-input" type={"radio"} name="rate"
+                    onChange={() => dispatch({ type: "RATINGS", payload: "3_above" })}
+                    checked={Pratings}
+                    ></input>
                     3 stars and above
                 </label>
                 <label htmlFor="rating-input">
-                    <input className="rating-input" type={"radio"}></input>
+                    <input className="rating-input" type={"radio"} name="rate"
+                    onChange={() => dispatch({ type: "RATINGS", payload: "2_above" })}
+                    checked={Pratings}
+                    ></input>
                     2 stars and above
                 </label>
                 <label htmlFor="rating-input">
-                    <input className="rating-input" type={"radio"}></input>
+                    <input className="rating-input" type={"radio"} name="rate"
+                    onChange={() => dispatch({ type: "RATINGS", payload: "1_above" })}
+                    checked={Pratings}
+                    ></input>
                     1 stars and above
```

Comment:
you can use map here for `[ {rating : 4, payload: "xascxczxas" } , . . . ] ` ratings, instead of reapeating code.

Reply:
Ok, thanks for suggestions!